testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java

```diff
@@ -77,4 +95,36 @@ protected void testCreateAndInsertTable(String schemaLocation)
             onTrino().executeQuery("DROP SCHEMA %1$s.test".formatted(getCatalogName()));
         }
     }
+
+    protected void testPathContainsSpecialCharacter(String schemaLocation, String partitioningPropertyName)
+    {
+        String tableName = "test_path_special_character" + randomNameSuffix();
+        try {
+            onTrino().executeQuery(format("CREATE SCHEMA %1$s.test WITH (location = '%2$s')", getCatalogName(), schemaLocation));
+            onTrino().executeQuery(format(
+                    "CREATE TABLE %1$s.test.%2$s (id bigint, part varchar) WITH (%3$s = ARRAY['part'])",
+                    getCatalogName(),
+                    tableName,
+                    partitioningPropertyName));
+
+            onTrino().executeQuery(format("INSERT INTO %1$s.test.%2$s VALUES " +
+                    "(1, 'with-hyphen')," +
+                    "(2, 'with.dot')," +
+                    "(3, 'with:colon')," +
+                    "(4, 'with/slash')," +
+                    "(5, 'with\\\\backslashes')," +
+                    "(6, 'with\\backslash')," +
+                    "(7, 'with=equal')," +
+                    "(8, 'with?question')," +
+                    "(9, 'with!exclamation')," +
+                    "(10, 'with%%percent')," +
```

Comment:
is it intentional that you use `%%` instead of `%` ?

Reply:
i don't follow what is happening here.  If we insert `(10, 'with%%percent')` why do we see `row(10, "with%percent")` when selecting?

Reply:
I didnt dig deep, but copied from TestTablePartitioningWithSpecialChars. Will try to find out

Reply:
This is inside of `String.format` method.

Reply:
done